### ISSUE TITLE
"No attr named 'NoOp' in NodeDef" error when running Tensorflow Transform in TFX

### ISSUE THREAD

**Issue by njunge94** [tensorflow/transform#224]
Hello TFT team, 
I am currently having trouble with the Tensorflow Transform running inside of a TFX Pipeline Transform component. I am copying over the issue template from TFX for more (hopefully helpful) information.

**System information** 
- Have I specified the code to reproduce the issue (Yes/No): Yes
- Environment in which the code is executed (e.g., Local
(Linux/MacOS/Windows), Interactive Notebook, Google Cloud, etc): Local MacOS Big Sur v11.2.1, standard Intel x86 arch.
 - TensorFlow version (you are using): 2.3.2
 - TFX Version: 0.26.1
 - Python version: 3.7.9
 - Tensorflow Transform version: 0.26.0

**Describe the current behavior**
I am trying to execute an NLP Pipeline. Most of my transform work is done in a component I wrote myself, but I am trying to include a dummy transform component to get a transform layer anyway (e.g. for model serving signatures with Tensorflow Serving).

My preprocessing_fn is trivial, as I do no real preprocessing here:
```
def preprocessing_fn(inputs):
    return inputs

```

Running the transform fails with the following traceback. It looks like it fails to find an operation called "NoOp" (which I assume is a placeholder) in a Transform Graph Node inside of TFT IO.

```
Traceback (most recent call last):
  File "/Users/nicholasjunge/workspaces/maiot/zenml/venv/lib/python3.7/site-packages/tensorflow/python/client/session.py", line 1365, in _do_call
    return fn(*args)
  File "/Users/nicholasjunge/workspaces/maiot/zenml/venv/lib/python3.7/site-packages/tensorflow/python/client/session.py", line 1348, in _run_fn
    self._extend_graph()
  File "/Users/nicholasjunge/workspaces/maiot/zenml/venv/lib/python3.7/site-packages/tensorflow/python/client/session.py", line 1388, in _extend_graph
    tf_session.ExtendSession(self._session)
tensorflow.python.framework.errors_impl.NotFoundError: No attr named 'NoOp' in NodeDef:
	 [[{{node transform/inputs/news/news}}]]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "apache_beam/runners/common.py", line 1239, in apache_beam.runners.common.DoFnRunner.process
  File "apache_beam/runners/common.py", line 768, in apache_beam.runners.common.PerWindowInvoker.invoke_process
  File "apache_beam/runners/common.py", line 893, in apache_beam.runners.common.PerWindowInvoker._invoke_process_per_window
  File "/Users/nicholasjunge/workspaces/maiot/zenml/venv/lib/python3.7/site-packages/apache_beam/transforms/core.py", line 1590, in <lambda>
    wrapper = lambda x, *args, **kwargs: [fn(x, *args, **kwargs)]
  File "/Users/nicholasjunge/workspaces/maiot/zenml/venv/lib/python3.7/site-packages/tensorflow_transform/beam/impl.py", line 782, in _infer_metadata_from_saved_model
    return _infer_metadata_from_saved_model_v1(saved_model_dir)
  File "/Users/nicholasjunge/workspaces/maiot/zenml/venv/lib/python3.7/site-packages/tensorflow_transform/beam/impl.py", line 796, in _infer_metadata_from_saved_model_v1
    session.run(tf.compat.v1.tables_initializer())
  File "/Users/nicholasjunge/workspaces/maiot/zenml/venv/lib/python3.7/site-packages/tensorflow/python/client/session.py", line 958, in run
    run_metadata_ptr)
  File "/Users/nicholasjunge/workspaces/maiot/zenml/venv/lib/python3.7/site-packages/tensorflow/python/client/session.py", line 1181, in _run
    feed_dict_tensor, options, run_metadata)
  File "/Users/nicholasjunge/workspaces/maiot/zenml/venv/lib/python3.7/site-packages/tensorflow/python/client/session.py", line 1359, in _do_run
    run_metadata)
  File "/Users/nicholasjunge/workspaces/maiot/zenml/venv/lib/python3.7/site-packages/tensorflow/python/client/session.py", line 1384, in _do_call
    raise type(e)(node_def, op, message)
tensorflow.python.framework.errors_impl.NotFoundError: No attr named 'NoOp' in NodeDef:
	 [[node transform/inputs/news/news (defined at Users/nicholasjunge/workspaces/maiot/zenml/venv/lib/python3.7/site-packages/tensorflow_transform/saved/saved_transform_io.py:271) ]]
```

Further information: `news` is the name of one of my input features.

**Describe the expected behavior**
The Transform runs, completes and saves a transform graph layer acting as the identity, i.e. acts functionally equivalent to `lambda x: x`.

I am a little bit puzzled about this, as other pipelines of mine work with very little preprocessing being done. I can share other preprocessing functions if requested. How can I make sure the transform graph gets understood correctly?

I attached the full traceback in a text file below.
[log.txt](https://github.com/tensorflow/tfx/files/6045418/log.txt)

**Comment by arghyaganguly** [comment]
@njunge94 , please provide the code for the custom transform component and other pre-processing functions to help understand more on this.
Thanks.

**Comment by njunge94** [comment]
Hello, 

I managed to solve the issue on accident. It turned out to be a fix on the TFX level. Sorry for posting here then, but I read the `tensorflow_transform` in the stack trace and thought the problem was due to the transform IO.

For anyone else running into this: If you work with a TFX `Transform` component like this, which runs Tensorflow Transform under the hood,

```
from tfx.components.transform.component import Transform
transform = Transform(
    preprocessing_fn=preprocessing_fn,
    examples=examples,
    schema=schema,
    custom_config=custom_config,
    force_tf_compat_v1=False,
)
```
try setting the `force_tf_compat_v1` flag to `False` like you see above. That is, if you are using Tensorflow 2.x code. For me, setting that flag to `False` explicitly (as of TFX 0.26.1, it is set to `True` by default) solved the problem.

Thank you for your patience. I am marking this as closed as it is resolved satisfactorily now.
